### ISSUE TITLE
[BUG] `from foo import *` imports should be ignored in stub files

### ISSUE THREAD

**Issue by AlexWaygood** [hadialqattan/pycln#152]
Hi!

I see `pycln` recently added support for `.pyi` files. That's awesome!

I'm interested in possibly using `pycln` in typeshed's CI. However, it seems like there's one part of [PEP 484](https://peps.python.org/pep-0484/#stub-files) that `pycln` isn't quite compliant with when it comes to stub files, namely this:

> [...] all objects imported into a stub using `from ... import *` are considered exported. (This makes it easier to re-export all objects from a given module that may vary by Python version.)

If I run `pycln` on typeshed at the moment, it erroneously deletes a large number of "unused" `from foo import *` imports. These imports are all deliberate -- if a type checker sees a `from foo import *` import in a stub file `bar.pyi`, it understands all symbols from module `foo` to be re-exported in module `bar`.

Playing around with `pycln`, it looks like this one-line patch solves the problem:

```diff
diff --git a/pycln/utils/refactor.py b/pycln/utils/refactor.py
index 4739e72..a241fc8 100644
--- a/pycln/utils/refactor.py
+++ b/pycln/utils/refactor.py
@@ -279,7 +279,7 @@ class Refactor:

                 # Expand any import '*' before checking.
                 node, is_star = self._expand_import_star(node)
-                if is_star is None:
+                if is_star is None or (is_star and self._path.is_stub):
                     continue

                 # Get set of used names.
```

I'd file a PR, but I'm struggling a little bit with the test suite 😅 (I'm not too familiar with the advanced features of pytest, I'm afraid.)

**Comment by hadialqattan** [comment]
Hi @AlexWaygood, this is a good catch.. actually, I planned to implement this within PR #150 but forgot to do so.

**Comment by hadialqattan** [comment]
@AlexWaygood after doing some investigation, your patch looks good to me, so please open a PR, and I'll help you finalize it even if you are unsure where/how to write the unit tests.

**Comment by hadialqattan** [comment]
\+ @AlexWaygood  I'm really happy to see you supporting the project. Thanks a lot bro!

**Comment by AlexWaygood** [comment]
> + @AlexWaygood I'm delighted to see you supporting the project. Thanks a lot bro!

Thank you for the very cool tool!

**Comment by hadialqattan** [comment]
@AlexWaygood I've just published a new release including the fix for this bug: [v2.0.2](https://hadialqattan.github.io/pycln/#/CHANGELOG?id=_202-2022-07-13).

**Comment by AlexWaygood** [comment]
> @AlexWaygood I've just published a new release including the fix for this bug: [v2.0.2](https://hadialqattan.github.io/pycln/#/CHANGELOG?id=_202-2022-07-13).

Awesome, thanks so much for your help!
